src/server/game/World/World.cpp

```diff
@@ -3351,17 +3351,27 @@ void World::AddDelayedDamage(Unit* attacker, Unit* victim, uint32 damage, CleanD
     delayedDamage.damageSchoolMask = damageSchoolMask;
     delayedDamage.spellProto = spellProto;
     delayedDamage.durabilityLoss = durabilityLoss;
+    delayedDamage.mapId = mapId;
+    delayedDamage.instanceId = instanceId;
     _delayedDamages.push_back(delayedDamage);
 }
 
 void World::ProcessDelayedDamages()
 {
     for (auto& damage : _delayedDamages)
     {
-        if (!damage.victim)
+        // Get map first
+        Map* map = sMapMgr->FindMap(damage.mapId, damage.instanceId);
+        if (!map)
             continue;
 
-        Unit::DealDamage(damage.attacker, damage.victim, damage.damage, damage.cleanDamage, damage.damagetype, damage.damageSchoolMask, damage.spellProto, damage.durabilityLoss);
+        // Now we get both, attacker and victim, but attacker can be null (although attacker is always a player).
+        Unit* attacker = ObjectAccessor::GetUnit(map, damage.attacker);
```

Comment:
dont need a nullcheck for attacker?

Reply:
No, attacker can be null on Unit::DealDamage